### PR TITLE
Servo

### PR DESCRIPTION
**Pull request by farhattameem**
Updated the Claw Mechanism, with a new branch.

### CODE REVIEW

File: hardware/Claw.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package org.firstinspires.ftc.teamcode.hardware;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.hardware.Servo;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+public class Claw extends Mechanism {
+    double servo0Position;
```
Comment (erxclau) [inline hardware/Claw.java]: no camel case

Comment (Lucas-Lee11) [inline hardware/Claw.java]: Redundant variable

File: hardware/Claw.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        servo0Position = 0.2;
+        servo0.setDirection(Servo.Direction.FORWARD);
+        servo0.setPosition(servo0Position);
+    }
+
+    public void closeClaw(){
+        servo0Position = 0.6;
+        servo0.setDirection(Servo.Direction.REVERSE);
+        servo0.setPosition(servo0Position);
+    }
```
Comment (Lucas-Lee11) [inline hardware/Claw.java]: Could change to return the position so that we don't have to hold to button

File: opmode/teleop/DriveTest.java [hunk 3]
```diff
@@ -3,31 +3,51 @@
...
+            double r = Math.hypot(gamepad1.left_stick_x, gamepad1.left_stick_y);
+            double robotAngle = Math.atan2(gamepad1.left_stick_y, gamepad1.left_stick_x) - Math.PI / 4;
+            double rightX = gamepad1.right_stick_x;
+
+            if(gamepad1.a) {
+                claw.openClaw();
+            }
+            if(gamepad1.b){
+                claw.closeClaw();
+            }
```
Comment (erxclau) [inline opmode/teleop/DriveTest.java]: brackets unnecessary

**Review by erxclau — CHANGES_REQUESTED** [review]

File: hardware/Drivetrain.java [hunk 4]
```diff
@@ -16,130 +17,15 @@
...
-            backLeft.setPower(0.4*inches/Math.abs(inches));
-        }
-
-        frontLeft.setPower(0.0);
-        backRight.setPower(0.0);
-        backLeft.setPower(0.0);
-        frontRight.setPower(0.0);
-
-    }
 }
```
Comment (Lucas-Lee11) [inline hardware/Drivetrain.java]: Need to keep

**Review by Lucas-Lee11 — APPROVED** [review]

File: hardware/Claw.java (hunk 2 above)
Comment (erxclau) [reply hardware/Claw.java]: what do you mean by hold to button

File: hardware/Claw.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package org.firstinspires.ftc.teamcode.hardware;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.hardware.Servo;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+public class Claw extends Mechanism {
+    private Servo servo0;
+    private Servo servo1;
```
Comment (erxclau) [inline hardware/Claw.java]: need better names

File: hardware/Claw.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    public Claw() {
+
+    }
```
Comment (erxclau) [inline hardware/Claw.java:12]: test without empty constructor

File: hardware/Claw.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+    public Claw(LinearOpMode opMode) {
+        this.opMode = opMode;
+    }
+
+    public void init(HardwareMap hwMap){
+        servo0 = hwMap.servo.get("servo0");
+        servo1 = hwMap.servo.get("servo1");
```
Comment (erxclau) [inline hardware/Claw.java]: better names

File: hardware/Claw.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    }
+    public void open(){
+        servo0.setDirection(Servo.Direction.FORWARD);
+        servo0.setPosition(0.2);
```
Comment (erxclau) [inline hardware/Claw.java]: generally a bad idea to have magic numbers?

File: hardware/Drivetrain.java [hunk 9]
```diff
@@ -43,27 +42,24 @@
...
 
-    public Servo servo;
     double  globalAngle, power = .30, correction;
     Orientation lastAngles = new Orientation();
 
     private double flPower = 0.0, frPower = 0.0, blPower = 0.0, brPower = 0.0;
 
-    public Drivetrain() {
-    }
+    public Drivetrain() {}
```
Comment (erxclau) [inline hardware/Drivetrain.java:50]: try deleting empty constructor

**Review by erxclau — CHANGES_REQUESTED** [review]

**Review by MattChann — APPROVED** [review]

**Pull request merged**